continuationsPlugin/src/main/scala/continuations/DefDefTransforms.scala

```diff
@@ -351,18 +379,108 @@ object DefDefTransforms extends TreesChecks:
         s.info.hasClassSymbol(requiredClass(suspendFullName)) || s.isTypeParam
       }
 
+    val newAnonFunctions = ListBuffer(transformedMethod.symbol)
     val substituteContinuation = new TreeTypeMap(
-      substFrom = List(parent) ++ oldMethodParamSymbols ++ contextFunctionOwner.toList,
-      substTo = List(transformedMethod.symbol) ++ transformedMethodParamSymbols ++
-        List(transformedMethod.symbol),
-      oldOwners = List(parent) ++ contextFunctionOwner.toList,
-      newOwners = List(transformedMethod.symbol, transformedMethod.symbol)
+      treeMap = {
+        case defdef: tpd.DefDef if defdef.symbol.isAnonymousFunction =>
+          val tpt: Type =
+            removeSuspendUpdateWithAny(defdef.tpt.tpe, updateWithAny = true)
+
+          val params: List[tpd.ParamClause] =
+            defdef
+              .paramss
+              .flatMap(_.flatMap {
+                case p: tpd.ValDef
+                    if p.symbol.info.hasClassSymbol(requiredClass(suspendFullName)) =>
+                  List.empty
+                case p: tpd.ValDef => List(List(p))
+                case p: tpd.TypeDef => List(List(p))
+              })
```

Comment:
```suggestion
          val params: List[tpd.ParamClause] = new TreeTypeMap(treeMap = {
            case p: tpd.ValDef
                if p.symbol.info.hasClassSymbol(requiredClass(suspendFullName)) =>
              tpd.EmptyTree
            case t: tpd.Tree => t
          }).transformParamss(defdef.paramss)

```

Reply:
Also, not necessary in this PR, but I'm pretty sure we can do almost all the things that have to do with substitution or replacing of symbols with the transforms on `TreeTypeMap`, which would make these a little less dense. I'll create a tech debt card for that.

Reply:
yes that's a good suggestion 👍

Reply:
done in [e2a341e](https://github.com/47deg/TBD/pull/75/commits/e2a341e48f06cda9f5bd1c5c54db13be16188387)